### PR TITLE
Make the tests faster, more reliable, and easier to extend

### PR DESCRIPTION
**Pull request by mjgiarlo**
# Why was this change made? 🤔

See above. Changes included:

* Remove `with_reindex` arg from reload_page_until_timeout helper
  * I added this ability a long time ago when indexing Fedora changes into Solr took more time, but I suspected it was no longer necessary and causes more clicks and uses more resources than is necessary. Upon removing this ability, I found that the test runtime is more or less unaffected.
* Move post-authentication timeout into configuration block
  * This commit moves a stray timeout into a block of similar timeouts in the application config. This keeps the configuration organized and consistent, making it easier to see all overrideable timeouts at a glance.
* Speed up and robustify ETD creation spec
  * This commit contains a number of changes related to improving the reliability and runtime speed of the ETD creation spec, including:
    * Reducing the time the test waits for negative expectations ("expect text NOT to appear"), where there is no need to wait longer than a second. Without this change, Capybara will wait until the default max wait time making the test take longer without any benefit(the page does not change underneath it, so waiting does not make sense in this context)
    * Dumbing down the embargo test somewhat, due to an occasional and annoying behavior where the embargo date is one day off. Dates and times are hard. Seeing this fail does not help anyone.
    * Forcing a reauthentication check when hitting Argo in the ETD spec. Without this, I was occasionally seeing behavior where ETD was authenticated but Argo was not, and moving over to Argo without going through the re-auth procedure caused a reproducible test timeout due to waiting on an unexpected Shib page.
* Refactor reload page helper
  * The `#reload_page_until_timeout!` method has become a bit of a beast whose behavior depends primarily on which arguments it is provided and the order the code checks those args. It is clear now that the best way to extend this method for all the different use cases it supports is to give it a single default behavior (check text on a page) and to allow consumers to provide a block for special behavior. This is a clearer way to extend Ruby methods to new use cases and is more Rubyish.

# Was README.md updated if necessary? 🤨

Yes.

### CODE REVIEW

File: spec/features/etd_creation_spec.rb
Comment (mjgiarlo) [inline spec/features/etd_creation_spec.rb:211]: I can be convinced that commenting this test out entirely is ill-advised, and that perhaps we should either

1. Check the embargo text at least exists: `reload_page_until_timeout!(text: 'Embargoed until')`; or
2. Check that either the embargoed date exists or the embarged date + one day exists, given the "bug" here is that the date is occasionally one day off (by being one day ahead of what's expected)

File: spec/support/page_helpers.rb
Comment (mjgiarlo) [inline spec/support/page_helpers.rb:4]: The way this method now works: either give me text, and I'll check the page for it, or give me a block and I'll yield to the consumer to do what it needs on each reload.

**Review by justinlittman — APPROVED** [review]

**Review by peetucket — APPROVED** [review]
I assume you've run the tests through a time or two with these changes.  If they all pass, thumbs up.

**Comment by mjgiarlo** [comment]
@peetucket I think it'd be good to hear how the test runs are going for @jmartin-sul this week before we get this merged, but I appreciate the review!

**Review by jmartin-sul — APPROVED** [review]
this worked pretty well for me, and felt at least as reliable as the `main` branch it's based on in a couple runs today.  i'm not sure what was happening earlier in the week, but when i ran the tests monday evening and tuesday morning, using both this branch and `main` at the time, i recall not getting a username/password prompt.  but i noticed running this and `main` again today, it was often just slow to appear after the browser came up.  perhaps i missed it, or had accidentally scrolled it off screen.

one comment about the commented out test, but wouldn't hold this up over that.

i did still occasionally get the weird shelving error (possible race condition) that #520 seems to have helped with.  but when i tested a branch that combined this branch with #521, and rebased on the latest `main`, i stopped running into that issue.

File: spec/features/etd_creation_spec.rb
Comment (jmartin-sul) [inline spec/features/etd_creation_spec.rb:206]: i like this more than what i did in #521

**Pull request merged**
